dom/src/main/scala/org/http4s/dom/WebSocketClient.scala

```diff
@@ -72,7 +72,11 @@ object WebSocketClient {
               cb(Right(ws))
             }
 
-            ws.onerror = e => cb(Left(js.JavaScriptException(e)))
+            ws.onerror = { e =>
+              ws.onmessage = null // prevent using dispatcher after shutdown
+              ws.onclose = null // ditto
+              cb(Left(js.JavaScriptException(e)))
+            }
```

Comment:
So I did some more investigation and I found out the `'error'` handler is kind of pointless.
1. it doesn't provide any detail about the error, _by-design_
    https://websockets.spec.whatwg.org/#eventdef-websocket-error
2. it's always followed by a `'close'` event (also by spec), so we might as well just use that

along these lines, I have a different proposal in 86bfb39e83d9b9ba93ace410961ae5e8d0a87860, wdyt?

Reply:
Great, one less callback to worry about! How shall we proceed?